engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtNDArrayEx.java

```diff
@@ -694,7 +694,54 @@ public NDList multiBoxPrior(
             List<Float> steps,
             List<Float> offsets,
             boolean clip) {
-        throw new UnsupportedOperationException("Not implemented");
+
+        NDManager ndManager = array.getManager().getParentManager();
+
+        Float stepX = steps.get(1);
+        Float stepY = steps.get(0);
+        int numSizes = sizes.size();
+        int numRatios = ratios.size();
+        int count = 0;
+
+        Shape input = array.getShape();
+        int inHeight = Math.toIntExact(input.get(2));
+        int inWidth = Math.toIntExact(input.get(3));
+
+        float[][] out = new float[inHeight * inWidth * numSizes * 2][4];
+
+        for (int r = 0; r < inHeight; ++r) {
+            float centerY = (r + offsets.get(0)) * stepY;
+            for (int c = 0; c < inWidth; ++c) {
+                float centerX = (c + offsets.get(1)) * stepX;
+                // ratio = first ratio, various sizes
+                float ratio = numRatios > 0 ? (float) Math.sqrt(ratios.get(0)) : 1.f;
+                for (int i = 0; i < numSizes; ++i) {
+                    float size = sizes.get(i);
+                    float w = size * inHeight / inWidth * ratio / 2;
+                    float h = size / ratio / 2;
+
+                    out[count][0] = centerX - w; // xmin
+                    out[count][1] = centerY - h; // ymin
+                    out[count][2] = centerX + w; // xmax
+                    out[count][3] = centerY + h; // ymax
+                    ++count;
+                }
+                // various ratios, size = min_size = size[0]
+                float size = sizes.get(0);
+                for (int j = 1; j < numRatios; ++j) {
+                    float ratioLocal = (float) Math.sqrt(ratios.get(j));
+                    float w = size * inHeight / inWidth * ratioLocal / 2;
+                    float h = size / ratioLocal / 2;
+                    out[count][0] = centerX - w; // xmin
+                    out[count][1] = centerY - h; // ymin
+                    out[count][2] = centerX + w; // xmax
+                    out[count][3] = centerY + h; // ymax
+                    ++count;
+                }
+            }
+        }
+        NDArray ndArray = ndManager.create(out).expandDims(0);
```

Comment:
I didn't look into your algorithm, but seems the final output is different from MXNet. Can you debug why it's different?

Reply:
There is some discussion of this here:
https://github.com/deepjavalibrary/djl/issues/2693

I'd love to debug it on the mxnet side, but this is non-trivial.  I'm working on a method to do so, but would appreciate some pointers.

Reply:
Thanks @frankfliu I found the bug and put in a fix. 

For me the first element in the result is `-0.09902344` in my implementation and mxnet. 
From your perspective is this now returning the correct values?